crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -799,6 +799,32 @@ where
                             return Some(Ok(()))
                         }
 
+                        // update the canon chain if continuous is enabled
+                        if self.sync.run_pipeline_continuously() {
+                            let max_block = ctrl.progress().unwrap_or_default();
+                            let max_hash = match self.blockchain.block_hash(max_block) {
+                                Ok(Some(block)) => block,
+                                Ok(None) => {
+                                    return Some(Err(Error::Provider(
+                                        ProviderError::HeaderNotFound(max_block.into()),
+                                    )
+                                    .into()))
+                                }
+                                Err(error) => return Some(Err(error.into())),
+                            };
+                            let max_header = match self.blockchain.block_by_number(max_block) {
```

Comment:
This is only the case when the pipeline finished, right, because we already update this on FCU?

this is a bit hard to follow and i'd like to perhaps move this to a separate function if possible with some docs.

Reply:
Yes, although we only update this while handling the FCU if we already have the blocks contained in the FCU.

Added a separate method